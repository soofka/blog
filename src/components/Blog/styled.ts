import styledComponents from 'styled-components';

export const StyledBlog = styledComponents.div`
  width: 100%;
  height: 100%;
  font-size: ${props => props.theme.baseFontSize};
  line-height: ${props => props.theme.baseLineHeight};
`;

export default StyledBlog;
