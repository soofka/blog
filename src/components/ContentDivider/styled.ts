import styledComponents from 'styled-components';

export const StyledContentDivider = styledComponents.hr`
  border-color: ${props => props.theme.lighterBlue} !important;
`;

export default StyledContentDivider;
