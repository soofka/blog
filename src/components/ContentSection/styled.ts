import styledComponents from 'styled-components';

export const StyledContentSection = styledComponents.div`
  float: left;
  margin: 0 0 0 ${props => props.theme.infoSectionWidth};
  padding: ${props => props.theme.paddingM};
  width: 100%;
  height: auto;
  min-height: 100%;
  box-sizing: border-box;
  color: ${props => props.theme.lighterBlack};
  background-color: ${props => props.theme.white};
  box-shadow: ${props => `0 0 ${props.theme.baseDistanceValue}px 0 ${props.theme.white}`};

  a {
    color: ${props => props.theme.lighterBlack};

    &:hover,
    &:active {
      color: ${props => props.theme.black};
    }
  }

@media (max-width: ${props => props.theme.mobileBreakPointWidth}) {
  margin: 0;
  padding: ${props => props.theme.paddingS};
  width: 100%;
}`;

export default StyledContentSection;
