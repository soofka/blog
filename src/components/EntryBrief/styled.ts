import styledComponents from 'styled-components';

export const StyledEntryBrief = styledComponents.div`
  float: left;
  width: 100%;
  height: ${props => 24 * props.theme.baseDistance}px;
  margin: ${props => props.theme.marginS} 0 ${props => props.theme.marginM} 0;
  padding: 0;
  border: 2px solid ${props => props.theme.white};
  outline: 1px solid ${props => props.theme.lighterBlack};
  background-repeat: no-repeat;
  background-position: center center;
  background-attachment: scroll;
  background-size: cover;
`;

export default StyledEntryBrief;
