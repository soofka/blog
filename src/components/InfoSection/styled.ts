import styledComponents from 'styled-components';

export const StyledInfoSection = styledComponents.div`
  position: fixed;
  top: 0;
  bottom: 0;
  box-sizing: border-box;
  width: ${props => props.theme.infoSectionWidth};
  height: auto;
  min-height: 100%;
  padding: ${props => props.theme.paddingS};
  color: ${props => props.theme.darkerWhite};
  background-color: ${props => props.theme.blue};
  box-shadow: ${props => `0 0 ${props.theme.baseDistance}px 0 ${props.theme.darkerBlue}`};

  a {
    color: ${props => props.theme.darkerWhite};
    text-decoration: none;

    &:hover,
    &:active {
      color: ${props => props.theme.white};
    }
  }

  @media (max-width: ${props => props.theme.mobileBreakPointWidth}) {
    position: static;
    display: inline-block;
    width: 100%;
    min-height: 0;
}
`;

export default StyledInfoSection;

