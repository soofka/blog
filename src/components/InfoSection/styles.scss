@import 'components/Blog/variables.scss';

.info-section {
  position: fixed;
  top: 0;
  bottom: 0;
  box-sizing: border-box;
  width: $infoSectionWidth;
  height: auto;
  min-height: 100%;
  padding: $paddingS;
  color: $darkerWhite;
  background-color: $blue;
  box-shadow: 0 0 $baseDistance 0 $darkerBlue;

  a {
    color: $darkerWhite;
    text-decoration: none;

    &:hover,
    &:active {
      color: $white;
    }
  }
}

@media (max-width: $mobileBreakPointWidth) {
  .info-section {
    position: static;
    display: inline-block;
    width: 100%;
    min-height: 0;
  }
}
