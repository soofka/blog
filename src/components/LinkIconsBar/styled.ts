import styledComponents from 'styled-components';

export const StyledLinkIconsBar = styledComponents.ul`
  margin: ${props => props.theme.marginM};
  padding: 0;
  text-align: center;

  li {
    display: inline-block;
    margin: ${props => `${props.theme.marginM} ${props.theme.marginXS}`};
    padding: 0;
    font-size: ${props => props.theme.fontSizeXXL};
    line-height: ${props => props.theme.fontSizeXXL};

    svg {
      width: 40px;
      height: 40px;
      opacity: 0.7;

      &:hover,
      &:active {
        opacity: 1;
      }
    }
  }

  @media (max-width: ${props => props.theme.mobileBreakPointWidth}) {
    float: right;
    margin: ${props => props.theme.marginS} 0;

    li {
      margin: 0 ${props => props.theme.marginXS};
      padding: 0;
      font-size: ${props => props.theme.fontSizeL};
      line-height: ${props => props.theme.fontSizeL};
    }
  }
`;

export default StyledLinkIconsBar;
