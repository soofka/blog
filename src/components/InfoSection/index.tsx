import * as React from 'react';
import { inject } from 'mobx-react';

import Logo from 'components/Logo';
import Quote from 'components/Quote';
import LinkIconsBar from 'components/LinkIconsBar';
import LanguageSwitcher from 'components/LanguageSwitcher';

import { LanguageStoreInterface } from 'store/language';

import './styles.scss';

interface InfoSectionPropsInterface {
  languageStore?: LanguageStoreInterface;
}

export const InfoSection = (props: InfoSectionPropsInterface): JSX.Element => {
  const { languageStore } = props;

  return (
    <div className="info-section">
      <Logo/>
      <LanguageSwitcher
        onLanguageSelection={(language: string) => languageStore.setLanguage(language)}
      />
      <Quote />
      <LinkIconsBar/>
    </div>
  );
};

export default inject('languageStore')(InfoSection);
