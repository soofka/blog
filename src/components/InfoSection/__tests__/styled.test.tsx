import * as React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import theme from 'common/theme';
import { StyledInfoSection } from 'components/InfoSection/styled';

describe('<StyledInfoSection/>', () => {

  it('is styled correctly', () => {
    expect(shallow(<StyledInfoSection theme={theme} />)).toMatchSnapshot();
  });

});
