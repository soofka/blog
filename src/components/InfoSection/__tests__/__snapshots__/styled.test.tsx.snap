// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledInfoSection/> is styled correctly 1`] = `
.c0 {
  position: fixed;
  top: 0;
  bottom: 0;
  box-sizing: border-box;
  width: 32%;
  height: auto;
  min-height: 100%;
  padding: 8px;
  color: rgb(198,198,198);
  background-color: rgb(0,1,49);
  box-shadow: 0 0 8px 0 rgb(0,0,0);
}

.c0 a {
  color: rgb(198,198,198);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:hover,
.c0 a:active {
  color: rgb(246,246,246);
}

@media (max-width:968px) {
  .c0 {
    position: static;
    display: inline-block;
    width: 100%;
    min-height: 0;
  }
}

<div
  className="c0"
/>
`;
